dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/ingest/UningestService.java

```
@@ -0,0 +1,189 @@
+
+    private DataAccess dataAccess;
+    private DataTableRepository dataTableRepository;
+    private IngestReportRepository ingestReportRepository;
+    private DataFileTagRepository dataFileTagRepository;
+    private DataFileRepository dataFileRepository;
+    private FileMetadataRepository fileMetadataRepository;
+    private MapLayerMetadataServiceBean mapLayerMetadataService;
+    private DatasetVersionRepository datasetVersionRepository;
+    private DatasetVersionServiceBean datasetVersionService;
+
+    // -------------------- CONSTRUCTORS --------------------
+
+    public UningestService() {
+        this.dataAccess = DataAccess.dataAccess();
+    }
+
+    @Inject
+    public UningestService(DataTableRepository dataTableRepository, IngestReportRepository ingestReportRepository,
+                           DataFileTagRepository dataFileTagRepository, DataFileRepository dataFileRepository,
+                           FileMetadataRepository fileMetadataRepository, MapLayerMetadataServiceBean mapLayerMetadataService,
+                           DatasetVersionRepository datasetVersionRepository, DatasetVersionServiceBean datasetVersionService) {
+        this();
+        this.mapLayerMetadataService = mapLayerMetadataService;
+        this.datasetVersionRepository = datasetVersionRepository;
+        this.dataTableRepository = dataTableRepository;
+        this.ingestReportRepository = ingestReportRepository;
+        this.dataFileTagRepository = dataFileTagRepository;
+        this.dataFileRepository = dataFileRepository;
+        this.fileMetadataRepository = fileMetadataRepository;
+        this.datasetVersionService = datasetVersionService;
+    }
+
+    // -------------------- LOGIC --------------------
+
+    public void uningest(List<DataFile> files, AuthenticatedUser user) {
+        if (files == null || files.isEmpty() || user == null) {
+            return;
+        }
```

IMO it is better to throw an exception instead of skipping execution.